### PR TITLE
Ensure `Section` has `[data-rac]` data attribute

### PR DESCRIPTION
**Pull request by amitdahan**
Related to #5748, follow-up to #6436 but for `Section`.

https://github.com/adobe/react-spectrum/pull/6436#issuecomment-2130061245

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 📝 Test Instructions:

Use `Section` within a `Menu` or `ListBox`, ensure the rendered `<section>` has the `data-rac` data attribute.

### CODE REVIEW

File: packages/react-aria-components/src/ListBox.tsx [hunk 1]
```diff
@@ -266,13 +266,18 @@ function ListBoxSection<T extends object>(props: SectionProps<T>, ref: Forwarded
     heading,
     'aria-label': props['aria-label'] ?? undefined
   });
+  let renderProps = useRenderProps({
+    defaultClassName: 'react-aria-Section',
```
Comment (snowystinger) [inline packages/react-aria-components/src/ListBox.tsx:270]:
```suggestion
    defaultClassName: 'react-aria-ListBoxSection',
```

File: packages/react-aria-components/src/Menu.tsx [hunk 2]
```diff
@@ -230,13 +230,18 @@ function MenuSection<T extends object>(props: SectionProps<T>, ref: ForwardedRef
     heading,
     'aria-label': section.props['aria-label'] ?? undefined
   });
+  let renderProps = useRenderProps({
+    defaultClassName: 'react-aria-Section',
```
Comment (snowystinger) [inline packages/react-aria-components/src/Menu.tsx:234]:
```suggestion
    defaultClassName: 'react-aria-MenuSection',
```

**Review by snowystinger — COMMENTED** [review]
Thanks for the PR. Items are prefixed by their component, we should do the same with Section

File: packages/react-aria-components/src/ListBox.tsx (hunk 1 above)
Comment (amitdahan) [reply packages/react-aria-components/src/ListBox.tsx:270]: hmmm, wouldn't that be a breaking-change?

Comment (snowystinger) [reply packages/react-aria-components/src/ListBox.tsx:270]: Good point, that's unfortunate
Maybe a good idea then to do `defaultClassName: 'react-aria-Section react-aria-ListBoxSection'`

Comment (amitdahan) [reply packages/react-aria-components/src/ListBox.tsx:270]: > as we don't actually have a ListBoxSection

My thought exactly!

I guess if we ever decide to have a `ListBoxSection` and `MenuSection` - then it'd make more sense?

Comment (snowystinger) [reply packages/react-aria-components/src/ListBox.tsx:270]: chatted with team, we can leave it as is until there is a need to add this

**Review by snowystinger — APPROVED** [review]

**Review by devongovett — APPROVED** [review]

**Pull request merged**